### ISSUE TITLE
uid should be username

### ISSUE THREAD

**Issue by unteem** [zorn-v/nextcloud-social-login#30]
In nextcloud the uid in the users table should actually be a username. 

Witth your app - great app by the way - after a users log in from an openid provider, he gets registered with an uid in the user table. Well it does make sense, but not in nextcloud world.

That uid is used for sharing, so it makes it difficult if its not the username, here you have to type the uid of the user to share a document.

I guess this should be a quick fix... I wish I could help but I don't know much about php...

thanks

**Comment by zorn-v** [comment]
> but not in nextcloud world.

You are wrong. it is ACTUALLY `UserID` 
It MUST be unique etc. Many things tied to uid (folder name for user on filesystem in data dir for example).

> here you have to type the uid of the user to share a document.

Or username (AFAIR) which you can change in personal settings

**Comment by unteem** [comment]
I agree that it must be unique etc like any userID but it looks like a username, otherwise with your app I end up sharing a file with someone called 7c93575b-6d50-4e88-b541-5a490c63ed57 and I cannot share it through his username. That does not work.

**Comment by zorn-v** [comment]
I think it because that user does not set username in nextcloud/external provider. In nextcloud username=uid if username not set.

**Comment by unteem** [comment]
Ok got it. It was my Identity Provider. I use keycloak and I had to change the mapping of the username attribute to the sub claim. (maybe in further releases, your app could offer to define that mapping. It could be a good option to work with any kind of OIDC providers)

But now, I have another issue :). I get a username that have this pattern: `internalNameOfOIDC-MYUSERNAME`. so in my case i have something like: indieID-username 

Can we get rid of the internal name of the custom open id connect endpoint we configured through your app?

Thanks again for this great app :)
